README.md

```diff
@@ -42,6 +42,55 @@ task assets build
 If Bash is available (Linux and Windows while on Git Bash), the commands will
 run in Bash, otherwise will fallback to `cmd` (on Windows).
 
+### Variables
+
+```yml
+build:
+  deps: [setvar]
+  cmds:
+  - echo "{{prefix}} {{THEVAR}}"
+  variables:
+    prefix: "Path:"
+
+setvar:
+  cmds:
+  - echo "{{PATH}}"
+  set: THEVAR
+```
+
+The above sample saves the path into a new variable which is then again echoed.
+
+You can use environment variables, task level variables and a file called `Variables` as source of variables.
+
+They are evaluated in the following order:
+
+Task local variables are overwritten by variables found in `Variables`. Variables found in `Variables` are overwritten with variables from the environment. The output of the last command is stored in the environment. So you can do something like this:
+
+```yml
+build:
+  deps: [setvar]
+  cmds:
+  - echo "{{prefix}} '{{THEVAR}}'"
```

Comment:
I think `prefix` should be `PREFIX`. Uppercase should be a convention for variables, just like environment variables.